tcms/testruns/views.py

```diff
@@ -63,57 +72,66 @@ def post(self, request):
-                    notes=form.cleaned_data.get('notes'),
-                    plan=test_plan,
-                    build=form.cleaned_data['build'],
-                    manager=form.cleaned_data['manager'],
-                    default_tester=default_tester,
-                )
-
-                loop = 1
-                for case in form.cleaned_data['case']:
-                    try:
-                        tcp = TestCasePlan.objects.get(plan=test_plan, case=case)
-                        sortkey = tcp.sortkey
-                    except ObjectDoesNotExist:
-                        sortkey = loop * 10
-
-                    test_run.add_case_run(case=case, sortkey=sortkey,
-                                          assignee=default_tester)
-                    loop += 1
-
-                return HttpResponseRedirect(
-                    reverse('testruns-get', args=[test_run.pk, ])
-                )
-
-        else:
-            form = NewRunForm(initial={
-                'summary': 'Test run for %s' % test_plan.name,
-                'manager': test_plan.author.email,
-                'default_tester': request.user.email,
-                'notes': '',
-            })
-            form.populate(product_id=test_plan.product_id)
+        form = NewRunForm(initial={
+            'summary': 'Test run for %s' % test_plan.name,
+            'manager': test_plan.author.email,
+            'default_tester': request.user.email,
+            'notes': '',
+        })
+        form.populate(product_id=test_plan.product_id)
 
         context_data = {
             'test_plan': test_plan,
             'test_cases': tcs_values,
             'form': form,
+            'disabled_cases': disabled_cases
         }
         return render(request, self.template_name, context_data)
 
+    def post(self, request):
+        # If from_plan does not exist will redirect to plans for select a plan
+        plan_id = request.POST.get('from_plan')
+        if not plan_id:
+            messages.add_message(request,
+                                 messages.ERROR,
+                                 _('Creating a TestRun requires a TestPlan, select one'))
+            return HttpResponseRedirect(reverse('plans-search'))
+
+        test_plan = TestPlan.objects.get(pk=plan_id)
+
+        form = NewRunForm(request.POST)
+        form.populate(product_id=test_plan.product_id)
+
+        if form.is_valid():
+            # Process the data in form.cleaned_data
+            default_tester = form.cleaned_data['default_tester']
+
+            test_run = TestRun.objects.create(
+                product_version=test_plan.product_version,
+                stop_date=None,
+                summary=form.cleaned_data.get('summary'),
+                notes=form.cleaned_data.get('notes'),
+                plan=test_plan,
+                build=form.cleaned_data['build'],
+                manager=form.cleaned_data['manager'],
+                default_tester=default_tester,
+            )
```

Comment:
NewRunForm should be decoupled from BaseRun form and only exclude many-to-many fields. Then you should be able to do form.save() instead of manually creating the new object.

Warning: this will also affect the clone view, but ultimately it needs to be refactored anyway. After the changes I have the feeling that clone view should inherit from the current one with minimal changes. (But that yields yet more changes).